praw/models/reddit/submission.py

```diff
@@ -681,4 +701,53 @@ def unspoiler(self):
         )
 
 
+class SubmissionModerationFlair(SubmissionFlair):
+    """Provides a set of functions allowing moderators to set post flairs."""
+
+    def choices(self):
+        """Return list of available flair choices.
+
+        Choices are required in order to use :meth:`.select`.
+
+        For example:
+
+        .. code:: python
+
+           choices = submission.mod.flair.choices()
+
+        """
+        return list(iter(self.submission.subreddit.link_templates))
```

Comment:
Is there a reason we're using `list(iter(...)))` here? What does `link_templates` return? If it's a generator, I'd prefer to not force it to a list.

Reply:
it returns a list of choices, so I forced it as a list

Reply:
although wait

Reply:
It is a generator but the superclass returns a list so i don't know if I should match the style or not